src/v2i-hub/CommandPlugin/src/CommandControl.cpp

```diff
@@ -129,114 +130,64 @@ void CommandPlugin::GetTelemetry(const string & dataType)
 	if (processTelemetry)
 	{
 		//clear updates
-		pluginsUpdatesJSON->clear();
+		if(dataType =="List"){
+			pluginsUpdatesContainer->clear();
+		}else{
+			pluginsUpdatesJSON->clear();
+		}		
 		//for each plugin
 		tmx::message_container_type *output = _tmxControl.GetOutput();
-		BOOST_FOREACH(ptree::value_type &plugin, output->get_storage().get_tree())
-		{
-			stringstream ss;
-			ss.clear();
-			ss.str(string());
-			//get plugin name
-			string pluginName = plugin.first;
-			//put name in list for later use
-			pluginNames.insert(pluginName);
-			//get plugin json
-			message_tree_type tmpTree;
-			if (!plugin.second.empty())
-				tmpTree.put_child(plugin.first, plugin.second);
-			else
-				tmpTree.put(plugin.first, plugin.second.data());
-
-			boost::property_tree::write_json(ss, tmpTree, false);
-			pJSON = ss.str();
-			boost::algorithm::trim(pJSON);
-			//compare json, if not found or different add to updates
-			bool found = false;
-			if (pluginsJSON->find(pluginName) != pluginsJSON->end())
-				found = true;
-			if (!found || (*pluginsJSON)[pluginName] != pJSON)
+		auto payloadExist = output->get_storage().get_tree().get_child_optional("payload");
+		if(payloadExist){
+			auto payload = payloadExist.get();
+			BOOST_FOREACH(ptree::value_type &payloadItem, payload)
```

Comment:
NIT: We should probably be removing BOOST_FOREACH calls and replacing them with C++ for each calls